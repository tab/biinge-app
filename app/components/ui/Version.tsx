import React from "react"
import { View } from "react-native"
import { useTranslation } from "react-i18next"

import Typography from "components/ui/Typography"
import { menuVersionStyles } from "styles"

const AppVersionComponent = () => {
  const { t } = useTranslation()

  return (
    <View style={menuVersionStyles.root}>
      <Typography variant="caption1" style={menuVersionStyles.text}>
        {t("app.version")}
      </Typography>
      <Typography variant="caption1" style={menuVersionStyles.text}>
        {t("app.name")}
      </Typography>
    </View>
  )
}

export default AppVersionComponent
