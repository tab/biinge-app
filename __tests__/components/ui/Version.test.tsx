import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import Version from "components/ui/Version"

describe("Version", () => {
  it("renders correctly", () => {
    const tree = render(<Version />)
    expect(tree).toMatchSnapshot()
  })
})
