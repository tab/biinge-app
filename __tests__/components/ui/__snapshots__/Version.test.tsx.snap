// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Version renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 14,
          "fontWeight": "700",
          "lineHeight": 17,
        },
        {
          "color": "#666666",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 24,
        },
      ]
    }
  >
    app.version
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={0}
    style={
      [
        {
          "color": "#181818",
          "fontSize": 14,
          "fontWeight": "700",
          "lineHeight": 17,
        },
        {
          "color": "#666666",
          "fontSize": 15,
          "fontWeight": "600",
          "lineHeight": 24,
        },
      ]
    }
  >
    app.name
  </Text>
</View>
`;
